bundles/org.openhab.binding.ring/src/main/java/org/openhab/binding/ring/handler/AccountHandler.java

```diff
@@ -0,0 +1,656 @@
+                    hardwareId);
+            saveRefreshTokenToFile(userProfile.getRefreshToken());
+        } catch (AuthenticationException ex) {
+            logger.debug("AuthenticationException when initializing Ring Account handler{}", ex.getMessage());
+            if (ex.getMessage().startsWith("Two factor")) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, ex.getMessage());
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, ex.getMessage());
+            }
+        } catch (JsonParseException e) {
+            logger.debug("Invalid response from api.ring.com when initializing Ring Account handler{}", e.getMessage());
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "Invalid response from api.ring.com");
+        }
+        logger.debug("doLogin RT: {}", getRefreshTokenFromFile());
+        try {
+            refreshRegistry();
+            updateStatus(ThingStatus.ONLINE);
+        } catch (DuplicateIdException ignored) {
+            updateStatus(ThingStatus.ONLINE);
+        } catch (AuthenticationException ae) {
+            registry = null;
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "AuthenticationException response from ring.com");
+            logger.debug("RestClient reported AuthenticationException in finally block: {}", ae.getMessage());
+        } catch (JsonParseException pe1) {
+            registry = null;
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "JsonParseException response from ring.com");
+            logger.debug("RestClient reported JsonParseException in finally block: {}", pe1.getMessage());
+        }
+    }
+
+    public @NonNullByDefault({}) String getHardwareId() {
+        AccountConfiguration config = getConfigAs(AccountConfiguration.class);
+        String hardwareId = config.hardwareId;
+        logger.debug("getHardwareId H:{}", hardwareId);
+        Configuration updatedConfiguration = getThing().getConfiguration();
+        try {
+            if (hardwareId.isEmpty()) {
+                hardwareId = getLocalMAC();
+                if (("".equals(hardwareId)) || hardwareId.isEmpty()) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                            "Hardware ID missing, check thing config");
+                    return hardwareId;
+                }
+                logger.debug("getHardwareId getLocalMac H:{}", hardwareId);
+                // write hardwareId to thing config
+                config.hardwareId = hardwareId;
+                updatedConfiguration.put("hardwareId", config.hardwareId);
+                updateConfiguration(updatedConfiguration);
+            }
+        } catch (Exception e) {
```

Comment:
Usually we try to avoid catching `Exception`. What do you expect here?

Reply:
It was due to getLocalMAC - which was throwing Exception - changed getLocalMAC to IOException